designdocs/batched_messages.md

```diff
@@ -10,92 +10,148 @@
+  [Avro](https://avro.apache.org/docs/current/) are supported.
+* `PayloadLocation` - The location of the payload. Can be either `OnChain` or `IPFS`.
 
-* `BatchAnnouncement`: implements `BatchAnnouncementParams`, returned by `get_batches`
+### Traits
+* `Model` - TBD. A set of functions for accessing message payload information.
+  * Derives `Encode`, `Decode`, `MaxEncodedLen`
+  * `max_length`: `SchemaMaxBytesBoundedVecLimit`
 
-See the [implementation of paging in the messages pallet](https://github.com/LibertyDSNP/mrc/blob/main/common/primitives/src/messages.rs#L26-L58) for comparison.
-
-* `BatchAnnouncementOptions<T:Config>`
-    * `msa_id`:  `Option<MsaId>`, the announcer's MSA id.  Pass None() to get all announcements.
-    * `from_block`: `<T::BlockNumber>`, retrieve messages starting at the given block number (inclusive)
-    * `to_block`: `<T::BlockNumber>`, retrieve messages ending at the given block number (inclusive)
-    * `from_index`: `u32`, starting message index
-    * `page_size`: `usize`, retrieve `page_size` messages at a time, up to configured `T::BatchPageSizeMax`. If 0, return `T::BatchPageSizeMax` results
-
-* `BatchAnnouncementResult`
-    * `has_next`: `uint32`, current page number
-    * `next_block`: `<T::BlockNumber>` starting block number of next page of results
-    * `next_index`: `u32` starting index of next results in `next_block`
-    * `results`: `Vec<BatchAnnouncement>`
+### Types
+* `SchemaAnnouncement<T:Config, M: Model>`: generic
+    * `model_type`: `ModelType` See enum section above.
+    * `model`: `M` Defines the shape of the message payload.
+    * `payload_location`: `PayloadLocation` See enum section above.
+
+* `MessageAnnouncement<T:Config>`: generic
+    * `schema_id`: `u16`
+    * `source`: `MsaId` Source of the message.
+    * `provider`: `MsaId` Public key of a capacity-providing account
+    * `onchain_payload`: `Vec<u8>?`
+    * `offchain_payload`: `Vec<u8>?`
 
 ### Extrinsics
-#### announce_batch(origin, batch_announcement_params)
-Creates and posts a new batch announcement message on chain, using the batch message Schema Id. This Schema Id must already be registered and will need to be fetched by the extrinsic.  The transaction fee is determined in part by the file size.
+#### register_schema(origin, schema_params)
+Creates and posts a new schema on chain. The transaction fee is determined in part by the model size.
 
 * **Parameters**
   * origin:  required for all extrinsics, the caller/sender.
-  * `batch_announcement_params`: `BatchAnnouncementParams`, the parameters to use in the batch announcement.
+  * `schema_params`: `SchemaAnnouncement`, the parameters to use in the batch announcement.
 
-* **Event**:  `Event::<T>::BatchAnnounced(schema_id, msa_id, file_size, batch_uri)`
+* **Event**:  `Event::<T>::SchemaAnnounced(schema_id, model, model_type, payload_location)`
```

Comment:
I don't think we have any new events other than what is already in the Message design doc.

SchemaCreated is already a thing, and when the messages design doc covers the events when a new message is created.